src/main/java/org/apache/gossip/crdt/GrowOnlyCounter.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.gossip.crdt;
+
+
+import java.lang.annotation.ElementType;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GrowOnlyCounter implements CrdtCounter<Long, GrowOnlyCounter> {
+
+
+    private final Map<String, Long> counters = new HashMap<>();
+    private final String myID;
+
+    public GrowOnlyCounter(String myID) {
+        this.myID = myID;
+        counters.putIfAbsent(myID, 0L);
+    }
+
+    private GrowOnlyCounter(String myID, Long count) {
+        this.myID = myID;
+        counters.putIfAbsent(myID, count);
+    }
+
+    private GrowOnlyCounter(String myID, Map<String, Long> counters) {
+        this.myID = myID;
+        this.counters.putAll(counters);
+    }
+
+
+    @Override
+    public GrowOnlyCounter merge(GrowOnlyCounter other) {
```

Comment:
For the other Crdt types they are immutable and merge returns a new instance. I would follow that pattern.